pkg/image/controller/imagestream_controller.go

```diff
@@ -238,38 +241,47 @@ func handleImageStream(
 		notifier.Importing(stream)
 	}
 
-	isi := &imageapi.ImageStreamImport{
+	isi := &imagev1.ImageStreamImport{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:            stream.Name,
 			Namespace:       stream.Namespace,
 			ResourceVersion: stream.ResourceVersion,
 			UID:             stream.UID,
 		},
-		Spec: imageapi.ImageStreamImportSpec{Import: true},
+		Spec: imagev1.ImageStreamImportSpec{Import: true},
 	}
-	for tag, tagRef := range stream.Spec.Tags {
+	for _, tagRef := range stream.Spec.Tags {
 		if tagImportable(tagRef) &&
-			(tagNeedsImport(stream, tag, tagRef, true) || !partial) {
-			isi.Spec.Images = append(isi.Spec.Images, imageapi.ImageImportSpec{
-				From:            kapi.ObjectReference{Kind: "DockerImage", Name: tagRef.From.Name},
-				To:              &kapi.LocalObjectReference{Name: tag},
+			(tagNeedsImport(stream, tagRef, true) || !partial) {
+			isi.Spec.Images = append(isi.Spec.Images, imagev1.ImageImportSpec{
+				From:            corev1.ObjectReference{Kind: "DockerImage", Name: tagRef.From.Name},
```

Comment:
do we need APIVersion here?

Reply:
For now we can get away without it, but we could add it to be fully API-compliant.